File: Project Euler/p12.py
# Highly divisible triangular number
from math import sqrt

def divisors(n):
    div = []
    for i in range(1, int(sqrt(n))+1):
        if n%i == 0:
            if (int(n/i) == i):
                div.append(i)
            else:
                div.append(i)
                div.append(int(n/i))
    return div

def main():
    triangle_num = 0
    counter = 1
    while True:
        triangle_num += counter
        if len(divisors(triangle_num)) >= 500:
            return triangle_num
        counter += 1

if __name__ == "__main__":
    print(main())

File: Project Euler/p12.cpp
// Highly divisible triangular number
#include <bits/stdc++.h>

using namespace std;

vector<int> divisors(int n) {
    vector<int> div;
    for (int i = 1; i <= sqrt(n); i++) {
        if (n%i == 0) {
            if (n/i == i) {
                div.push_back(i);
            } else {
                div.push_back(i);
                div.push_back(n/i);
            }
        }
    }
    return div;
}

int main() {
    int triangle_num = 0;
    int counter = 1;
    while (true) {
        triangle_num += counter;
        if (divisors(triangle_num).size() >= 500) {
            cout << triangle_num << endl;
            return 0;
        }
        counter ++;
    }
}
